### PR TITLE
feat(DropdownMenu): Add `DropdownMenu.Trigger`

### LINKED ISSUE #1415: Add `Trigger` sub-component to `DropdownMenu`

### LINKED ISSUE #1437: Update `DropdownMenu` documentation
Update `DropdownMenu` documentation to better reflect new changes in #1415. 

Follow the defined template for documentation defined in #777. 

Define the chapters we have information on.

For example, we need "Bruk" chapter with examples on how to use `.Trigger` and `anchorEl` as an alternative.

### PR DESCRIPTION
**Pull request by Barsnes**
Resolves #1415
Resolves #1437

### CODE REVIEW

**Comment by Barsnes** [comment]
Followed the same pattern as in `Popover` #1406.
Tried to keep the implementation and docs the same, for easier understanding.

**Comment by mimarz** [comment]
Blæææ, opposed to `Popover` this is a big breaking change for users already using `DropdowMenu`... We kinda wanted to start get used to giving people headsup before doing this.

**Comment by mimarz** [comment]
> > Blæææ, opposed to `Popover` this is a big breaking change for users already using `DropdowMenu`... We kinda wanted to start get used to giving people headsup before doing this.
> 
> 😅 Yeah, thankfully they would just need to wrap their content in a `Dropdown.Content` component. Maybe we can give them a search and replace they can copy to fix it? Should be able to get `<DropdownMenu {anything}>` -> `<DropdownMenu><DropdownMenu.Content>` `</DropdownMenu>` -> `</DropdownMenu.Content></DropdownMenu>`

yeah, either write it in the release msg on Slack or maybe a short chapter on it in the Storybook page when doing #1437

File: packages/react/src/components/DropdownMenu/DropdownMenu.mdx [hunk 1]
```diff
@@ -11,34 +11,66 @@ import { DropdownMenu } from './';
...
+
+Legg på `portal` for å rendere `DropdownMenu` i en portal. Dette flytter dropdownen til `body`.
 
 <Canvas of={DropdownMenuStories.InPortal} />
 
-### `Dropdown.Group`
+## Tilgjengelighet
+
+Dersom du bruker `DropdownMenu.Trigger` er det innebyd tilgjengelighet for deg, men dersom du bruker `anchorEl` må du styre tilgjengelighet selv.
+Da må du legge på `aria-expanded={true/false}` og `aria-haspopup='menu'`.
```
Comment (mimarz) [inline packages/react/src/components/DropdownMenu/DropdownMenu.mdx:53]: This was a very long sentence. Suggestion:

```suggestion
Brukes `DropdownMenu.Trigger` er det innebygd tilgjengelighet.

Brukes `anchorEl` må du sørge for å legge på `aria-expanded={true/false}` i henhold til åpne/lukket tilstand og `aria-haspopup='menu'` på trigger elementet.
```

File: packages/react/src/components/DropdownMenu/DropdownMenu.mdx [hunk 2]
```diff
@@ -11,34 +11,66 @@ import { DropdownMenu } from './';
...
+
+### `anchorEl`
+
+_Dette vil bli fjernet i fremtiden, og vi anbefaler å bruke `DropdownMenu.Trigger` istedenfor._
+
+Du kan bruke `anchorEl` istedenfor `DropdownMenu.Trigger` for å styre hvor `DropdownMenu` skal vises, men da må du styre åpning og lukking av `DropdownMenu` selv.
+Vi anbefaler å bruke `DropdownMenu.Trigger` istedenfor `anchorEl` fordi det er bygd inn tilgjenelighetshåndtering for deg.
```
Comment (mimarz) [inline packages/react/src/components/DropdownMenu/DropdownMenu.mdx:60]: Suggest moving this under "Slik bruker du `DropdownMenu`" and changing the title to "Alternativ"

Comment (mimarz) [inline packages/react/src/components/DropdownMenu/DropdownMenu.mdx:60]: A bit much repetition. Suggestion:

```suggestion

Alternativt kan du bruke `anchorEl` for å styre hvor `DropdownMenu` skal vises, men da må du selv håndtere tilstand for åpning/lukking og attributter for tilgjengelighet.

_`anchorEl` vil bli fjernet i fremtiden, og vi anbefaler å bruke `DropdownMenu.Trigger` istedenfor._
```

File: packages/react/src/components/DropdownMenu/DropdownMenu.mdx [hunk 3]
```diff
@@ -11,34 +11,66 @@ import { DropdownMenu } from './';
...
+
+### `DropdownMenu.Group`
 
 <ArgsTable of={DropdownMenu.Group} />
 
-## Referanser
+### `DropdownMenu.Trigger`
+
+Med `DropdownMenu.Trigger` kan du styre hvordan triggeren til `DropdownMenu` skal vises. Du kan bruke asChild for å endre hvordan `DropdownMenu.Trigger` rendres.
+Dersom du skal legge på funksjoner som `onClick`, legg det på ditt element, og legg asChild på `DropdownMenu.Trigger`. Triggeren rendrer `Button` som standard.
```
Comment (mimarz) [inline packages/react/src/components/DropdownMenu/DropdownMenu.mdx:71]:
```suggestion
Triggeren er en [Button](/docs/felles-button--docs) som standard.

Bruk `DropdownMenu.Trigger` til å aktivere `DropdownMenu`. Du kan bruke `asChild` for å endre `DropdownMenu.Trigger` elementet.
Dersom du skal legge på funksjoner som `onClick`, legg det på ditt element, og legg `asChild` på `DropdownMenu.Trigger`. 
```

File: packages/react/src/components/DropdownMenu/DropdownMenu.tsx [hunk 4]
```diff
@@ -44,101 +23,63 @@ export type DropdownMenuProps = {
...
-      open,
-      onOpenChange: () => onClose && onClose(),
-      elements: {
-        reference: anchorEl,
-        floating: floatingEl.current,
-      },
-      whileElementsMounted: autoUpdate,
-      middleware: [offset(GAP), shift()],
-    });
+  const anchor = anchorEl ?? triggerRef.current;
```
Comment (mimarz) [inline packages/react/src/components/DropdownMenu/DropdownMenu.tsx:41]: This should be called `anchorEl`

File: packages/react/src/components/DropdownMenu/DropdownMenuContent.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    portal,
+    anchor,
+    isControlled,
+    internalOpen,
+    setInternalOpen,
+    onClose,
+  } = useContext(DropdownMenuContext);
+
+  const Container = portal ? FloatingPortal : React.Fragment;
+  const floatingEl = useRef<HTMLUListElement>(null);
```
Comment (mimarz) [inline packages/react/src/components/DropdownMenu/DropdownMenuContent.tsx:45]: Might aswell fix this also -> `floatingRef`

**Review by mimarz — CHANGES_REQUESTED** [review]
Good stuff ☕ 

Feedback is mostly docs stuff and some const names!

File: packages/react/src/components/DropdownMenu/DropdownMenuContent.tsx (hunk 5 above)
Comment (Barsnes) [reply packages/react/src/components/DropdownMenu/DropdownMenuContent.tsx:45]: `Popover` uses `floatingEl`, so I think we should use this here as well

Comment (mimarz) [reply packages/react/src/components/DropdownMenu/DropdownMenuContent.tsx:45]: Aight, lets fix it in a chore afterwards 👍

**Review by mimarz — APPROVED** [review]
Nice work! 👍

**Pull request merged**